From f8e51a0b6d84eff9d421b9119bc71c56bb34d6d6 Mon Sep 17 00:00:00 2001
From: Arnaldo Carvalho de Melo <acme@redhat.com>
Date: Wed, 3 Jul 2013 12:43:03 -0300
Subject: [PATCH 587/858] Revert "bnx2x: set gso_type"

This reverts commit 0aba93e2b9fb0be4f4dd1b52ef10d789edf15f74.

This was just a minimal fix for 3.8, but as we're backporting all
the way to 3.9.4, just revert it and we will get the full fix.

Signed-off-by: Arnaldo Carvalho de Melo <acme@redhat.com>
---
 drivers/net/ethernet/broadcom/bnx2x/bnx2x_cmn.c | 12 +++++++-----
 1 file changed, 7 insertions(+), 5 deletions(-)

diff --git a/drivers/net/ethernet/broadcom/bnx2x/bnx2x_cmn.c b/drivers/net/ethernet/broadcom/bnx2x/bnx2x_cmn.c
index 9cd2277..25bb320 100644
--- a/drivers/net/ethernet/broadcom/bnx2x/bnx2x_cmn.c
+++ b/drivers/net/ethernet/broadcom/bnx2x/bnx2x_cmn.c
@@ -530,11 +530,13 @@ static int bnx2x_fill_frag_skb(struct bnx2x *bp, struct bnx2x_fastpath *fp,
   skb_shinfo(skb)->gso_size = bnx2x_set_lro_mss(bp,
      tpa_info->parsing_flags, len_on_bd);
 
-  skb_shinfo(skb)->gso_type =
-   (GET_FLAG(tpa_info->parsing_flags,
-      PARSING_FLAGS_OVER_ETHERNET_PROTOCOL) ==
-    PRS_FLAG_OVERETH_IPV6) ?
-   SKB_GSO_TCPV6 : SKB_GSO_TCPV4;
+  /* set for GRO */
+  if (fp->mode == TPA_MODE_GRO)
+   skb_shinfo(skb)->gso_type =
+       (GET_FLAG(tpa_info->parsing_flags,
+          PARSING_FLAGS_OVER_ETHERNET_PROTOCOL) ==
+      PRS_FLAG_OVERETH_IPV6) ?
+    SKB_GSO_TCPV6 : SKB_GSO_TCPV4;
  }
 
 
-- 
1.8.3.1